src/main/java/com/squareup/kotlinpoet/jvm/ClassNames.kt

```diff
@@ -0,0 +1,73 @@
+/*
+ * Copyright (C) 2017 Square, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+@file:JvmName("ClassNamesJvm")
```

Comment:
Ideally that should remain as "ClassNames", but there's the `KClass<*>.asClassName()` function that stays in the main module and Java clients that need methods from both versions will have to use fully-qualified class names. Maybe there's a better way.

Reply:
How about `@JvmMultifileClass` and the same `@JvmName`?

Reply:
I doubt it'll work across multiple modules, but I can give it a try

Reply:
There won't be multiple modules. You'll have the common module with common code and then the jvm module will implement the common module and add JVM-specific things. There will only be one class file in the final artifact. And then for JS there will be a JS module, etc., and they're free to have platform-specific things as well.

Reply:
Oh, I see. Didn't think about it from the perspective of cross-platform modules, rather that `kotlinpoet-jvm` will just provide extensions for `kotlinpoet`. I'm curious to try the cross-platform approach.

Reply:
Yeah the nice thing about using multiplatform rather than multiple modules is that you pick one artifact based on the platform on which your code generator runs. And if your code generator is multiplatform, then you only get the common stuff.

If we can do multifile classes for now let's do that. If we end up doing two artifacts it's an easy change to make.